### PR TITLE
Frontend support for configurable record summaries

### PR DESCRIPTION
**Pull request by seancolsen**
⚠️ Do not merge until #3997 is merged.

Fixes: #3890

## Demo

https://github.com/user-attachments/assets/db446add-f3a0-42ef-946f-e20f6513116b

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
Ok this is working pretty well now, and I'd like to start review.

@ghislaineguerin I've assigned you because I'd like you to be the first to review. I want to make sure we're agreed on the UX before moving on to any code-level review. Watch the demo video or play around with it locally. I'm happy to hop on a call to chat more if you like. Once you feel good about it, then I'll open it up to Pavish for code review.

@kgodey I you might like to watch the demo video too in case you have any critique of the new UX here.

**Comment by seancolsen** [comment]
Thanks @ghislaineguerin. 

@pavish I'd like you to review this now. Keep in mind it's stacked on #3997 which you're also assigned to review.

File: mathesar_ui/src/systems/data-explorer/utils.ts [hunk 1]
```diff
@@ -465,52 +457,46 @@ export function speculateColumnMetaData({
...
+                      // TODO_BETA: Ask Pavish.
+                      //
+                      // `Column['type_options']` was previously typed loosely
+                      // as `Record<string, unknown> | null`. Now it's more
+                      // strict and it doesn't have a `type` property.
+                      //
+                      // type:
+                      //   updatedColumnsMetaData.get(aggregation.inputAlias)
+                      //     ?.column.type ?? 'unknown',
+                    }
```
Comment (pavish) [inline mathesar_ui/src/systems/data-explorer/utils.ts:494]: We discussed this on a 1:1 before. I think we missed making the code change.

`type` here is now changed to `item_type`. This applies for `_array` types which require the type of items they hold.

**Review by pavish — APPROVED** [review]
@seancolsen This looks great! Nice work!

I haven't reviewed the code in detail, I mostly scanned for lines that stood out and they all seem fine.

**Pull request merged**